lib/sequin/accounts/user.ex

```diff
@@ -23,14 +32,40 @@ defmodule Sequin.Accounts.User do
     field :auth_provider_id, :string
     field :extra, :map
 
-    belongs_to :account, Account
+    has_many :accounts_users, AccountUser
+    has_many :accounts, through: [:accounts_users, :account]
     field :impersonating_account, :map, virtual: true
 
     timestamps(type: :utc_datetime)
   end
 
   def where_account_id(query \\ base_query(), account_id) do
-    from([user: u] in query, where: u.account_id == ^account_id)
+    from([user: u] in query,
+      join: ua in AccountUser,
+      on: ua.user_id == u.id,
+      where: ua.account_id == ^account_id
+    )
+  end
+
+  def current_account(%User{} = user) do
+    user =
+      if Ecto.assoc_loaded?(user.accounts_users) && Ecto.assoc_loaded?(user.accounts) do
+        user
+      else
+        Sequin.Repo.preload(user, [:accounts_users, :accounts])
+      end
```

Comment:
```suggestion
    user = Sequin.Repo.preload(user, [:accounts_users, :accounts])
```

Repo.preload will do nothing if they are already loaded